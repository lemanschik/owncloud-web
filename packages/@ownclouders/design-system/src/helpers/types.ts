export type IconFillType = 'fill' | 'line' | 'none'
