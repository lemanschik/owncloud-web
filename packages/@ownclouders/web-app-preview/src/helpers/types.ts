export type CachedFile = {
  id: string
  name: string
  url: string
  ext: string
  mimeType: string
  isVideo: boolean
  isImage: boolean
  isAudio: boolean
}
