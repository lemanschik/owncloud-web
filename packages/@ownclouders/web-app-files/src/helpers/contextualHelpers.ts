import { ConfigurationManager } from 'web-pkg'
import { omit } from 'lodash-es'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

export interface ContextualHelperDataListItem {
  text: string
  headline?: boolean
}
export interface ContextualHelperData {
  title: string
  text?: string
  list?: ContextualHelperDataListItem[]
  readMoreLink?: string
}

export interface ContextualHelperOptions {
  configurationManager: ConfigurationManager
}

export const shareInviteCollaboratorHelp = (options: ContextualHelperOptions) =>
  filterContextHelper(
    {
      title: $gettext('Share with people'),
      text: $gettext(
        'Use the input field to search for users and groups. Select them to share the item.'
      ),
      list: [
        { text: $gettext('Subfolders'), headline: true },
        {
          text: $gettext(
            'If you share a folder, all of its contents and subfolders will be shared as well.'
          )
        },
        { text: $gettext('Notification'), headline: true },
        {
          text: $gettext(
            'People you share resources with will be notified via email or in-app notification.'
          )
        },
        { text: $gettext('Incognito'), headline: true },
        {
          text: $gettext('People you share resources with can not see who else has access.')
        },
        { text: $gettext('“via folder”'), headline: true },
        {
          text: $gettext(
            'The “via folder” is shown next to a share, if access has already been given via a parent folder. Click on the “via folder” to edit the share on its parent folder.'
          )
        }
      ],
      readMoreLink: 'https://doc.owncloud.com/webui/next/owncloud_web/web_for_users.html#sharing'
    },
    options
  )

export const shareInviteCollaboratorHelpCern = (options: ContextualHelperOptions) =>
  filterContextHelper(
    {
      title: '',
      list: [
        { text: $gettext('Search for service or secondary Account'), headline: true },
        {
          text: $gettext(
            'To search for service or secondary accounts prefix the username with "a:" (like "a:doe") and for guest accounts prefix the username with "l:" (like "l:doe").'
          )
        }
      ]
    },
    options
  )

export const shareSpaceAddMemberHelp = (options: ContextualHelperOptions) =>
  filterContextHelper(
    {
      title: $gettext('Add members to this Space'),
      text: $gettext('Enter a name to add people or groups as members to this Space.'),
      list: [
        { text: $gettext('Member capabilities'), headline: true },
        {
          text: $gettext(
            'Members are able to see who has access to this space and access all files in this space. Read or write permissions can be set by assigning a role.'
          )
        },
        { text: $gettext('Space manager capabilities'), headline: true },
        {
          text: $gettext(
            'Members with the Manager role are able to edit all properties and content of a Space, such as adding or removing members, sharing subfolders with non-members, or creating links to share.'
          )
        }
      ],
      readMoreLink: 'https://doc.owncloud.com/webui/next/owncloud_web/web_for_users.html#sharing'
    },
    options
  )
export const shareViaLinkHelp = (options: ContextualHelperOptions) =>
  filterContextHelper(
    {
      title: $gettext('Choose how access is granted'),
      list: [
        {
          text: $gettext('Only invited people can access'),
          headline: true
        },
        {
          text: $gettext(
            'Account and login is required. Only people from the list "Invited people" can access.'
          )
        },
        {
          text: $gettext('Everyone with the link'),
          headline: true
        },
        {
          text: $gettext(
            'No login required. Everyone with the link can access. If you share this link with people from the list "Invited people", they need to login so that their individual assigned permissions can take effect. If they are not logged-in, the permissions of the link take effect.'
          )
        }
      ],
      readMoreLink: 'https://doc.owncloud.com/webui/next/owncloud_web/web_for_users.html#sharing'
    },
    options
  )
export const shareViaIndirectLinkHelp = (options: ContextualHelperOptions) =>
  filterContextHelper(
    {
      title: $gettext('What are indirect links?'),
      text: $gettext('Indirect links are links giving access by a parent folder.'),
      list: [
        {
          text: $gettext('How to edit indirect links'),
          headline: true
        },
        {
          text: $gettext(
            'Indirect links can only be edited in their parent folder. Click on the folder icon below the link to navgate to the parent folder.'
          )
        }
      ],
      readMoreLink: 'https://doc.owncloud.com/webui/next/owncloud_web/web_for_users.html#sharing'
    },
    options
  )

const filterContextHelper = (
  data: ContextualHelperData,
  options?: ContextualHelperOptions
): ContextualHelperData => {
  if (options.configurationManager.options.contextHelpersReadMore === false) {
    return omit(data, 'readMoreLink')
  }
  return data
}
