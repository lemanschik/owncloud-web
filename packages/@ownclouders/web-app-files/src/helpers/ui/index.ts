export * as fileList from './filesList'
export * as resourceTable from './resourceTable'
export * as resourceTiles from './resourceTiles'
