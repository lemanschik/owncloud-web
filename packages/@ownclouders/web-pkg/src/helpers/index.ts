export * from './cache'
export * from './contextMenuDropdown'
export * from './download'
export * from './datetime'
export * from './filesize'
export * from './fuse'
export * from './locale'
