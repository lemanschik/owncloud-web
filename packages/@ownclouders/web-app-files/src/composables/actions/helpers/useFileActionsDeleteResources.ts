import { Store } from 'vuex'
import { cloneStateObject } from '../../../helpers/store'
import { isSameResource } from '../../../helpers/resource'
import { buildWebDavFilesTrashPath } from '../../../helpers/resources'
import { buildWebDavSpacesTrashPath, Resource, SpaceResource } from 'web-client/src/helpers'
import PQueue from 'p-queue'
import { isLocationTrashActive, isLocationSpacesActive } from '../../../router'
import { dirname } from 'path'
import { createFileRouteOptions } from 'web-pkg/src/helpers/router'
import { computed, unref } from 'vue'
import {
  useCapabilitySpacesEnabled,
  useCapabilityShareJailEnabled,
  useClientService,
  useRouter,
  useStore,
  useLoadingService
} from 'web-pkg/src/composables'
import { useGettext } from 'vue3-gettext'
import { ref } from 'vue'

export const useFileActionsDeleteResources = ({ store }: { store?: Store<any> }) => {
  store = store || useStore()
  const router = useRouter()
  const language = useGettext()
  const { $gettext, $ngettext, interpolate: $gettextInterpolate } = language
  const hasShareJail = useCapabilityShareJailEnabled()
  const hasSpacesEnabled = useCapabilitySpacesEnabled()
  const clientService = useClientService()
  const loadingService = useLoadingService()
  const { owncloudSdk } = clientService

  const queue = new PQueue({ concurrency: 4 })
  const deleteOps = []
  const resourcesToDelete = ref([])

  const isInTrashbin = computed(() => {
    return isLocationTrashActive(router, 'files-trash-generic')
  })

  const resources = computed(() => {
    return cloneStateObject(unref(resourcesToDelete))
  })

  const dialogTitle = computed(() => {
    const currentResources = unref(resources)
    const isFolder = currentResources[0].type === 'folder'
    let title = null

    if (currentResources.length === 1) {
      if (isFolder) {
        title = unref(isInTrashbin)
          ? $gettext('Permanently delete folder %{name}')
          : $gettext('Delete folder %{name}')
      } else {
        title = unref(isInTrashbin)
          ? $gettext('Permanently delete file %{name}')
          : $gettext('Delete file %{name}')
      }
      return $gettextInterpolate(
        title,
        {
          name: currentResources[0].name
        },
        true
      )
    }

    title = unref(isInTrashbin)
      ? $ngettext(
          'Permanently delete selected resource?',
          'Permanently delete %{amount} selected resources?',
          currentResources.length
        )
      : $ngettext(
          'Delete selected resource?',
          'Delete %{amount} selected resources?',
          currentResources.length
        )

    return $gettextInterpolate(title, { amount: currentResources.length }, false)
  })

  const dialogMessage = computed(() => {
    const currentResources = unref(resources)
    const isFolder = currentResources[0].type === 'folder'

    if (currentResources.length === 1) {
      if (isFolder) {
        return unref(isInTrashbin)
          ? $gettext(
              'Are you sure you want to delete this folder? All its content will be permanently removed. This action cannot be undone.'
            )
          : $gettext('Are you sure you want to delete this folder?')
      }
      return unref(isInTrashbin)
        ? $gettext(
            'Are you sure you want to delete this file? All its content will be permanently removed. This action cannot be undone.'
          )
        : $gettext('Are you sure you want to delete this file?')
    }

    return unref(isInTrashbin)
      ? $gettext(
          'Are you sure you want to delete all selected resources? All their content will be permanently removed. This action cannot be undone.'
        )
      : $gettext('Are you sure you want to delete all selected resources?')
  })

  const trashbin_deleteOp = (space, resource) => {
    const path = unref(hasShareJail)
      ? buildWebDavSpacesTrashPath(space.id)
      : buildWebDavFilesTrashPath(store.getters.user.id)

    return owncloudSdk.fileTrash
      .clearTrashBin(path, resource.id)
      .then(() => {
        store.dispatch('Files/removeFilesFromTrashbin', [resource])
        const translated = $gettext('"%{file}" was deleted successfully')
        store.dispatch('showMessage', {
          title: $gettextInterpolate(translated, { file: resource.name }, true)
        })
      })
      .catch((error) => {
        if (error.statusCode === 423) {
          // TODO: we need a may retry option ....
          const p = queue.add(() => {
            return trashbin_deleteOp(space, resource)
          })
          deleteOps.push(p)
          return
        }

        console.error(error)
        const translated = $gettext('Failed to delete "%{file}"')
        store.dispatch('showMessage', {
          title: $gettextInterpolate(translated, { file: resource.name }, true),
          status: 'danger'
        })
      })
  }

  const trashbin_delete = (space: SpaceResource) => {
    // TODO: use clear all if all files are selected
    store.dispatch('toggleModalConfirmButton')
    // FIXME: Implement proper batch delete and add loading indicator
    for (const file of unref(resources)) {
      const p = queue.add(() => {
        return trashbin_deleteOp(space, file)
      })
      deleteOps.push(p)
    }

    Promise.all(deleteOps).then(() => {
      store.dispatch('hideModal')
      store.dispatch('toggleModalConfirmButton')
    })
  }

  const filesList_delete = (space: SpaceResource) => {
    return loadingService.addTask(
      (loadingCallbackArgs) => {
        return store
          .dispatch('Files/deleteFiles', {
            ...language,
            space,
            files: unref(resources),
            clientService,
            loadingCallbackArgs
          })
          .then(async () => {
            store.dispatch('hideModal')
            store.dispatch('toggleModalConfirmButton')

            // Load quota
            if (
              isLocationSpacesActive(router, 'files-spaces-generic') &&
              !['public', 'share'].includes(space?.driveType)
            ) {
              if (unref(hasSpacesEnabled)) {
                const graphClient = clientService.graphAuthenticated
                const driveResponse = await graphClient.drives.getDrive(
                  unref(resources)[0].storageId
                )
                store.commit('runtime/spaces/UPDATE_SPACE_FIELD', {
                  id: driveResponse.data.id,
                  field: 'spaceQuota',
                  value: driveResponse.data.quota
                })
              } else {
                const user = await owncloudSdk.users.getUser(store.getters.user.id)
                store.commit('SET_QUOTA', user.quota)
              }
            }

            if (
              unref(resourcesToDelete).length &&
              isSameResource(unref(resourcesToDelete)[0], store.getters['Files/currentFolder'])
            ) {
              return router.push(
                createFileRouteOptions(space, {
                  path: dirname(unref(resourcesToDelete)[0].path),
                  fileId: unref(resourcesToDelete)[0].parentFolderId
                })
              )
            }
          })
      },
      { indeterminate: false }
    )
  }

  const deleteHelper = (space: SpaceResource) => {
    store.dispatch('toggleModalConfirmButton')

    unref(isInTrashbin) ? trashbin_delete(space) : filesList_delete(space)
  }

  const displayDialog = (space: SpaceResource, resources: Resource[]) => {
    resourcesToDelete.value = [...resources]

    const modal = {
      variation: 'danger',
      title: unref(dialogTitle),
      message: unref(dialogMessage),
      cancelText: $gettext('Cancel'),
      confirmText: $gettext('Delete'),
      onCancel: () => {
        store.dispatch('hideModal')
      },
      onConfirm: () => {
        deleteHelper(space)
      }
    }

    store.dispatch('createModal', modal)
  }

  return {
    displayDialog,
    // HACK: exported for unit tests:
    filesList_delete
  }
}
