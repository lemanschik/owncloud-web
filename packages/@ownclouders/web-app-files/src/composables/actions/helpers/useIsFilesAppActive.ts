import { computed, unref } from 'vue'
import { activeApp, useRoute } from 'web-pkg/src/composables'

const isFilesAppActive = (activeApp: string): boolean => {
  // FIXME: we should use this constant but it somehow breaks the unit tests
  // return activeApp === FilesApp.appInfo.id
  return activeApp === 'files'
}

export const useIsFilesAppActive = () => {
  const currentRoute = useRoute()

  return computed(() => isFilesAppActive(activeApp(unref(currentRoute))))
}
