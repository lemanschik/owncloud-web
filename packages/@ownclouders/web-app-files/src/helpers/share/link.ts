import { DateTime } from 'luxon'
import {
  LinkShareRoles,
  Share,
  linkRoleInternalFolder,
  linkRoleViewerFolder
} from 'web-client/src/helpers/share'
import { Store } from 'vuex'
import { ClientService } from 'web-pkg/src/services'
import { useClipboard } from '@vueuse/core'
import { Ability } from 'web-pkg'
import { Resource } from 'web-client'
import { Language } from 'vue3-gettext'

export interface CreateQuicklink {
  clientService: ClientService
  language: Language
  store: Store<any>
  storageId?: any
  resource: Resource
  password?: string
  ability: Ability
}

export const createQuicklink = async (args: CreateQuicklink): Promise<Share> => {
  const { clientService, resource, store, password, language, ability } = args
  const { $gettext } = language

  const canCreatePublicLink = ability.can('create-all', 'PublicLink')
  const allowResharing = store.state.user.capabilities.files_sharing?.resharing
  const capabilitiesRoleName =
    store.state.user.capabilities.files_sharing?.quick_link?.default_role ||
    linkRoleViewerFolder.name
  const canEdit = store.state.user.capabilities.files_sharing?.public?.can_edit || false
  const canContribute = store.state.user.capabilities.files_sharing?.public?.can_contribute || false
  const alias = store.state.user.capabilities.files_sharing?.public?.alias
  const roleName = !canCreatePublicLink
    ? linkRoleInternalFolder.name
    : capabilitiesRoleName || linkRoleViewerFolder.name
  const permissions = LinkShareRoles.getByName(
    roleName,
    resource.isFolder,
    canEdit,
    canContribute,
    alias
  ).bitmask(allowResharing)
  const params: { [key: string]: unknown } = {
    name: $gettext('Link'),
    permissions: permissions.toString(),
    quicklink: true
  }

  if (password) {
    params.password = password
  }

  const expirationDate = store.state.user.capabilities.files_sharing.public.expire_date

  if (expirationDate.enforced) {
    params.expireDate = DateTime.now()
      .plus({ days: parseInt(expirationDate.days, 10) })
      .endOf('day')
      .toISO()
  }

  // needs check for enforced password for default role (viewer?)
  // and concept to what happens if it is enforced

  params.spaceRef = resource.fileId || resource.id

  const link = await store.dispatch('Files/addLink', {
    path: resource.path,
    client: clientService.owncloudSdk,
    params,
    storageId: resource.fileId || resource.id
  })

  const { copy } = useClipboard({ legacy: true })
  copy(link.url)

  await store.dispatch('showMessage', {
    title: $gettext('The link has been copied to your clipboard.')
  })

  return link
}
