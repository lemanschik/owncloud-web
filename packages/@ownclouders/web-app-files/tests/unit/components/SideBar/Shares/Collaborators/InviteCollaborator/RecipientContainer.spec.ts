import RecipientContainer from 'web-app-files/src/components/SideBar/Shares/Collaborators/InviteCollaborator/RecipientContainer.vue'
import { ShareTypes } from 'web-client/src/helpers/share'
import { createStore, defaultPlugins, mount, defaultStoreMockOptions } from 'web-test-helpers'

jest.mock('web-app-files/src/helpers/user/avatarUrl', () => ({
  avatarUrl: jest.fn().mockReturnValue('avatarUrl')
}))

const getRecipient = (shareType: number | string = ShareTypes.user.value) => ({
  label: 'Albert Einstein',
  value: {
    shareType,
    shareWith: 'einstein',
    shareWithAdditionalInfo: 'einstein@example.org'
  }
})

describe('InviteCollaborator RecipientContainer', () => {
  describe('renders a recipient with a deselect button', () => {
    it.each(ShareTypes.authenticated)('different recipients for different shareTypes', (type) => {
      const recipient = getRecipient(type.key)
      const { wrapper } = getMountedWrapper(recipient)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
  it('displays an avatar image if capability is present', async () => {
    const recipient = getRecipient()
    const { wrapper } = getMountedWrapper(recipient, true)
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('emits an event if deselect button is clicked', async () => {
    const recipient = getRecipient()
    const { wrapper } = getMountedWrapper(recipient, true)
    const spyOnDeselect = wrapper.vm.deselect.mockImplementation()
    const button = wrapper.find('.files-share-invite-recipient-btn-remove')
    await button.trigger('click')
    expect(spyOnDeselect).toHaveBeenCalledTimes(1)
  })
})

function getMountedWrapper(recipient, avatarsEnabled = false) {
  const storeOptions = defaultStoreMockOptions
  storeOptions.getters.user.mockImplementation(() => ({ id: 1 }))
  storeOptions.getters.capabilities.mockImplementation(() => ({
    files_sharing: {
      user: {
        profile_picture: avatarsEnabled,
        expire_date: {
          enabled: true,
          days: 10
        }
      }
    }
  }))
  const store = createStore(storeOptions)
  return {
    wrapper: mount(RecipientContainer, {
      props: {
        recipient,
        deselect: jest.fn()
      },
      global: {
        plugins: [...defaultPlugins(), store],
        stubs: {
          OcIcon: true
        }
      }
    })
  }
}
