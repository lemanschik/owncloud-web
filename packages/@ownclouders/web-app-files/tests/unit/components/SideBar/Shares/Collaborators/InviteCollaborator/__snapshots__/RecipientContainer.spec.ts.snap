// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteCollaborator RecipientContainer displays an avatar image if capability is present 1`] = `
<span class="oc-recipient files-share-invite-recipient">
  <span aria-hidden="true" class="vue-avatar--wrapper oc-avatar oc-recipient-avatar" data-test-user-name="Albert Einstein" focusable="false" style="width: 16.8px; height: 16.8px; line-height: 16.8px; font-size: 6px; font-family: Helvetica, Arial, sans-serif; color: white;" width="16.8">
    <img alt="" aria-hidden="true" class="avatarImg" loading="lazy" src="avatarUrl">
  </span>
  <p class="oc-recipient-name">Albert Einstein</p> <!-- @slot Append content (actions, additional info, etc.)  -->
  <button aria-label="Deselect Albert Einstein" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw files-share-invite-recipient-btn-remove" type="button">
    <!-- @slot Content of the button -->
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="close" size="small" type="span" variation="passive"></oc-icon-stub>
  </button>
</span>
`;

exports[`InviteCollaborator RecipientContainer renders a recipient with a deselect button different recipients for different shareTypes 1`] = `
<span class="oc-recipient files-share-invite-recipient">
  <oc-icon-stub accessiblelabel="User" class="oc-recipient-icon" color="" filltype="fill" name="user" size="small" type="span" variation="passive"></oc-icon-stub>
  <p class="oc-recipient-name">Albert Einstein</p> <!-- @slot Append content (actions, additional info, etc.)  -->
  <button aria-label="Deselect Albert Einstein" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw files-share-invite-recipient-btn-remove" type="button">
    <!-- @slot Content of the button -->
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="close" size="small" type="span" variation="passive"></oc-icon-stub>
  </button>
</span>
`;

exports[`InviteCollaborator RecipientContainer renders a recipient with a deselect button different recipients for different shareTypes 2`] = `
<span class="oc-recipient files-share-invite-recipient">
  <oc-icon-stub accessiblelabel="User" class="oc-recipient-icon" color="" filltype="fill" name="user" size="small" type="span" variation="passive"></oc-icon-stub>
  <p class="oc-recipient-name">Albert Einstein</p> <!-- @slot Append content (actions, additional info, etc.)  -->
  <button aria-label="Deselect Albert Einstein" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw files-share-invite-recipient-btn-remove" type="button">
    <!-- @slot Content of the button -->
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="close" size="small" type="span" variation="passive"></oc-icon-stub>
  </button>
</span>
`;

exports[`InviteCollaborator RecipientContainer renders a recipient with a deselect button different recipients for different shareTypes 3`] = `
<span class="oc-recipient files-share-invite-recipient">
  <oc-icon-stub accessiblelabel="User" class="oc-recipient-icon" color="" filltype="fill" name="user" size="small" type="span" variation="passive"></oc-icon-stub>
  <p class="oc-recipient-name">Albert Einstein</p> <!-- @slot Append content (actions, additional info, etc.)  -->
  <button aria-label="Deselect Albert Einstein" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw files-share-invite-recipient-btn-remove" type="button">
    <!-- @slot Content of the button -->
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="close" size="small" type="span" variation="passive"></oc-icon-stub>
  </button>
</span>
`;

exports[`InviteCollaborator RecipientContainer renders a recipient with a deselect button different recipients for different shareTypes 4`] = `
<span class="oc-recipient files-share-invite-recipient">
  <oc-icon-stub accessiblelabel="User" class="oc-recipient-icon" color="" filltype="fill" name="user" size="small" type="span" variation="passive"></oc-icon-stub>
  <p class="oc-recipient-name">Albert Einstein</p> <!-- @slot Append content (actions, additional info, etc.)  -->
  <button aria-label="Deselect Albert Einstein" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw files-share-invite-recipient-btn-remove" type="button">
    <!-- @slot Content of the button -->
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="close" size="small" type="span" variation="passive"></oc-icon-stub>
  </button>
</span>
`;

exports[`InviteCollaborator RecipientContainer renders a recipient with a deselect button different recipients for different shareTypes 5`] = `
<span class="oc-recipient files-share-invite-recipient">
  <oc-icon-stub accessiblelabel="User" class="oc-recipient-icon" color="" filltype="fill" name="user" size="small" type="span" variation="passive"></oc-icon-stub>
  <p class="oc-recipient-name">Albert Einstein</p> <!-- @slot Append content (actions, additional info, etc.)  -->
  <button aria-label="Deselect Albert Einstein" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw files-share-invite-recipient-btn-remove" type="button">
    <!-- @slot Content of the button -->
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="close" size="small" type="span" variation="passive"></oc-icon-stub>
  </button>
</span>
`;

exports[`InviteCollaborator RecipientContainer renders a recipient with a deselect button different recipients for different shareTypes 6`] = `
<span class="oc-recipient files-share-invite-recipient">
  <oc-icon-stub accessiblelabel="User" class="oc-recipient-icon" color="" filltype="fill" name="user" size="small" type="span" variation="passive"></oc-icon-stub>
  <p class="oc-recipient-name">Albert Einstein</p> <!-- @slot Append content (actions, additional info, etc.)  -->
  <button aria-label="Deselect Albert Einstein" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw files-share-invite-recipient-btn-remove" type="button">
    <!-- @slot Content of the button -->
    <oc-icon-stub accessiblelabel="" color="" filltype="fill" name="close" size="small" type="span" variation="passive"></oc-icon-stub>
  </button>
</span>
`;
